cpp/ql/test/library-tests/ir/ir/raw_ir.expected

```diff
@@ -2666,64 +2666,67 @@ ir.cpp:
-#  577|     r0_30(glval<char>)    = PointerAdd[1]            : r0_27, r0_29
-#  577|     r0_31(char)           = Constant[0]              : 
-#  577|     mu0_32(char)          = Store                    : &:r0_30, r0_31
-#  577|     r0_33(int)            = Constant[1]              : 
-#  577|     r0_34(glval<char>)    = PointerAdd[1]            : r0_27, r0_33
-#  577|     r0_35(char)           = Constant[0]              : 
-#  577|     mu0_36(char)          = Store                    : &:r0_34, r0_35
-#  578|     r0_37(glval<char[2]>) = VariableAddress[e]       : 
-#  578|     mu0_38(char[2])       = Uninitialized[e]         : &:r0_37
-#  578|     r0_39(int)            = Constant[0]              : 
-#  578|     r0_40(glval<char>)    = PointerAdd[1]            : r0_37, r0_39
-#  578|     r0_41(char)           = Constant[0]              : 
-#  578|     mu0_42(char)          = Store                    : &:r0_40, r0_41
-#  578|     r0_43(int)            = Constant[1]              : 
-#  578|     r0_44(glval<char>)    = PointerAdd[1]            : r0_37, r0_43
-#  578|     r0_45(char)           = Constant[1]              : 
-#  578|     mu0_46(char)          = Store                    : &:r0_44, r0_45
-#  579|     r0_47(glval<char[3]>) = VariableAddress[f]       : 
-#  579|     mu0_48(char[3])       = Uninitialized[f]         : &:r0_47
-#  579|     r0_49(int)            = Constant[0]              : 
-#  579|     r0_50(glval<char>)    = PointerAdd[1]            : r0_47, r0_49
-#  579|     r0_51(char)           = Constant[0]              : 
-#  579|     mu0_52(char)          = Store                    : &:r0_50, r0_51
-#  579|     r0_53(int)            = Constant[1]              : 
-#  579|     r0_54(glval<char>)    = PointerAdd[1]            : r0_47, r0_53
-#  579|     r0_55(unknown[2])     = Constant[0]              : 
-#  579|     mu0_56(unknown[2])    = Store                    : &:r0_54, r0_55
-#  580|     v0_57(void)           = NoOp                     : 
-#  571|     v0_58(void)           = ReturnVoid               : 
-#  571|     v0_59(void)           = UnmodeledUse             : mu*
-#  571|     v0_60(void)           = AliasedUse               : ~mu0_2
-#  571|     v0_61(void)           = ExitFunction             : 
+#  572|     mu0_4(char[32])       = Uninitialized[a_pad]     : &:r0_3
+#  572|     r0_5(glval<char[1]>)  = StringConstant[""]       : 
+#  572|     r0_6(char[1])         = Load                     : &:r0_5, ~mu0_2
+#  572|     mu0_7(char[1])        = Store                    : &:r0_3, r0_6
+#  572|     r0_8(unknown[31])     = Constant[0]              : 
+#  572|     r0_9(int)             = Constant[1]              : 
+#  572|     r0_10(glval<char>)    = PointerAdd[1]            : r0_3, r0_9
+#  572|     mu0_11(unknown[31])   = Store                    : &:r0_10, r0_8
+#  573|     r0_12(glval<char[4]>) = VariableAddress[a_nopad] : 
+#  573|     mu0_13(char[4])       = Uninitialized[a_nopad]   : &:r0_12
```

Comment:
`a_nopad` and `a_infer` shouldn't need an `Uninitialized`, because the single `Store` covers the entire variable. Can you tighten the condition to only include string literal initialization where the variable size is greater than the string literal size?

Reply:
Done. It now depends on the `zeroInitRange` predicate that `TranslatedStringLiteralInitialization` uses to decide whether to insert padding bytes.